packages/server/src/automations/automationUtils.ts

```diff
@@ -114,12 +118,15 @@ export async function sendAutomationAttachmentsToStorage(
       attachmentRows[prop] = value
     }
   }
-
   for (const [prop, attachments] of Object.entries(attachmentRows)) {
-    if (attachments.length) {
-      row[prop] = await Promise.all(
-        attachments.map(attachment => generateAttachmentRow(attachment))
-      )
+    if (Array.isArray(attachments)) {
+      if (attachments.length) {
+        row[prop] = await Promise.all(
+          attachments.map(attachment => generateAttachmentRow(attachment))
+        )
+      }
+    } else if (Object.keys(row[prop]).length > 0) {
+      row[prop] = await generateAttachmentRow(attachments)
```

Comment:
We probably need a test in place to prevent the near-miss we had here happening again 🙏

Reply:
done!